Primo.js

```diff
@@ -251,48 +253,36 @@ function importPNX(text) {
 	item.title = ZU.xpathText(doc, '//display/title');
 	if(item.title) item.title = ZU.unescapeHTML(item.title);
 	
-	var creators;
-	var contributors;
-	if(ZU.xpathText(doc, '//display/creator')) {
-		creators = ZU.xpath(doc, '//display/creator'); 
-	}
-	
-	if(ZU.xpathText(doc, '//display/contributor')) {
-		contributors = ZU.xpath(doc, '//display/contributor'); 
-	}
-	
-	if(!creators && contributors) { // <creator> not available using <contributor> as author instead
+	var creators = ZU.xpath(doc, '//display/creator');
+	var contributors = ZU.xpath(doc, '//display/contributor');
+	if(!creators.length && contributors.length) {
+		// <creator> not available using <contributor> as author instead
 		creators = contributors;
-		contributors = null;
-	}
-	
-	if(!creators && ! contributors){
-		creators = ZU.xpath(doc, '//addata/addau')
+		contributors = [];
 	}
 	
-	for(i in creators) {
-		if(creators[i]) {
-			var creator  = ZU.unescapeHTML(creators[i].textContent).split(/\s*;\s*/);
-			for(j in creator){
-				creator[j] = creator[j].replace(/\d{4}-(\d{4})?/g, '');
-				item.creators.push(Zotero.Utilities.cleanAuthor(creator[j], "author", true));
-			}			
+	// //addata/addau is great because it lists authors in last, first format,
+	// but it can also have a bunch of junk. We'll use it to help split authors
+	var splitGuidance = {};
+	var addau = ZU.xpath(doc, '//addata/addau');
+	for (var i=0; i<addau.length; i++) {
+		var author = cleanAuthor(addau[i].textContent);
+		if (author.indexOf(',') != -1) {
+			var splitAu = author.split(',');
+			if (splitAu.length > 2) continue;
+			var name = splitAu[1].trim().toLowerCase() + ' '
+				+ splitAu[0].trim().toLowerCase()
+			splitGuidance[name] = author;
 		}
 	}
 	
-	for(i in contributors) {
-		if(contributors[i]) {
-			var contributor = ZU.unescapeHTML(contributors[i].textContent).split(/\s*;\s*/);
-			for(j in contributor){
-			contributor[j] = contributor[j].replace(/\d{4}-(\d{4})?/g, '');
-			item.creators.push(Zotero.Utilities.cleanAuthor(contributor[j], "contributor", true));
-			}			
-		}
-	}
+	fetchCreators(item, creators, 'author', splitGuidance);
+	fetchCreators(item, contributors, 'contributor', splitGuidance);
+	
 	//PNX doesn't do well with institutional authors; This isn't perfect, but it helps:
-	for(i in item.creators){
-		if(!item.creators[i].firstName){
-			item.creators[i].fieldMode=1;
+	for(var i=0; i<item.creators.length; i++) {
+		if(!item.creators[i].firstName) {
+			item.creators[i].fieldMode = 1;
```

Comment:
You are doing this in the `fetchCreators` function already, i.e. delete it here.